starter_code/src/movies.js

```diff
@@ -1,20 +1,118 @@
 /* eslint no-restricted-globals: 'off' */
-// Turn duration of the movies from hours to minutes 
+// Turn duration of the movies from hours to minutes
+// How to change the value of the same key in all objects of an array
 
+function turnHoursToMinutes(movies) {
+  var time = movies.map(function(oneMovie) {
+    var A = {};
+    var hoursIntoMinutes = Number(oneMovie.duration.charAt(0) * 60);
+    var minutes = Number(
+      oneMovie.duration.charAt(3) + oneMovie.duration.charAt(4)
+    );
+    console.log(hoursIntoMinutes + minutes);
+    return (oneMovie.duration = hoursIntoMinutes + minutes);
+  });
+  return time;
+}
 
-// Get the average of all rates with 2 decimals 
+console.log(turnHoursToMinutes(movies));
 
+// Get the average of all rates with 2 decimals
+
+function ratesAverage(movies) {
+  var average = movies.reduce(function(sum, oneMovie) {
+    return sum + Number(oneMovie.rate);
+  }, 0);
+  return Number((average / movies.length).toFixed(2));
+}
+
+console.log(ratesAverage(movies));
 
 // Get the average of Drama Movies
 
+function dramaMoviesRate(movies) {
+  var dramaMovies = movies.filter(function(oneMovie) {
+    return oneMovie.genre.indexOf("Drama") > -1;
+  });
+  if (dramaMovies.length === 0) {
+    return undefined;
+  }
+  return ratesAverage(dramaMovies);
+}
+
+console.log(dramaMoviesRate(movies));
 
 // Order by time duration, in growing order
+// Will do when Iteration 1 is solved
 
+// Order by date
+//Cannot solve the 2 Jasmine errors :
+// - Return the new array in growing order
+// -If two movies have the same length, order them alphabetically by their title
```

Comment:
the reason your tests are not passing is because you're ordering by year (Nizar said to do that)  rather than duration. But your code works :)